contents/tutorials/nuxt-analytics.md

```diff
@@ -0,0 +1,285 @@
+
+```bash
+npm install posthog-node
+```
+
+Next, initialize `posthog-node` wherever you'd like to capture events server-side, such as in your [server routes](https://nuxt.com/docs/guide/directory-structure/server) or in the [middleware](https://nuxt.com/docs/guide/directory-structure/server#server-middleware). For this tutorial, we'll show you how to capture an event in the middleware:
+
+Create a folder `middleware` in `server` directory and then a new file `analytics.js` in it:
+
+```bash
+cd ./server
+mkdir middleware
+cd ./middleware
+touch analytics.js
+```
+
+Inside `analytics.js`, create a middleware function. Initialize PostHog in it and capture an event:
+
+```js file=server/middleware/analytics.js
+import { PostHog } from 'posthog-node';
+
+export default defineEventHandler(async (event) => {
+  const runtimeConfig = useRuntimeConfig();
+  const posthog = new PostHog(
+    runtimeConfig.public.posthogPublicKey,
+    { host: runtimeConfig.public.posthogHost }
+  );
+
+  posthog.capture({
+    distinctId: 'placeholder_distinct_id_of_the_user', 
+    event: 'in_the_middleware',
+  });
+  await posthog.shutdownAsync()  
+});
+```
+
+> **Note**: Make sure to _always_ call `posthog.shutdownAsync()` after capturing events from the server-side.
+> PostHog queues events into larger batches, and this call forces all batched events to be flushed immediately.
+
+If you run your app again, you should begin to see `in_the_middleware` events in PostHog.
+
+### Handling `distinctId` on the server
+
+You may have noticed that we set `distinctId: placeholder_distinct_id_of_the_user` in our event above. To attribute events to the correct user, `distinctId` should be set to your user's unique id. For logged-in users, we typically use their email as their `distinctId`. However, for logged-out users, you can use the `distinct_id` property from their PostHog cookie:
+
+```js file=server/middleware/analytics.js
+import { PostHog } from 'posthog-node';
+
+export default defineEventHandler(async (event) => {
+  const cookieString = event.req.headers.cookie || '';  
+  const cookieName =`ph_${runtimeConfig.public.posthogPublicKey}_posthog`;
+  const cookieMatch = cookieString.match(new RegExp(cookieName + '=([^;]+)'));
+  
+  let distinctId;
+  if (cookieMatch) {
+    const parsedValue = JSON.parse(decodeURIComponent(cookieMatch[1]));
+    if (parsedValue && parsedValue.distinct_id) {
+      distinctId = parsedValue.distinct_id;
+      const runtimeConfig = useRuntimeConfig();
+      const posthog = new PostHog(
+        runtimeConfig.public.posthogPublicKey,
+        { host: runtimeConfig.public.posthogHost }
+      );
+      posthog.capture({
+        distinctId: distinctId, 
+        event: 'in_the_middleware',
+      });
+      await posthog.shutdownAsync()
+    } 
+  }
+});
+```
+
```

Comment:
Please let me know your feedback on the following:

In other similar tutorials, like [Remix](https://posthog.com/tutorials/remix-analytics), or [Astro](https://posthog.com/tutorials/astro-analytics), when we use `crypto` to generate the distinct id on the server, we then the topic of `identify` to merge the server and the client users.

I've excluded it from this tutorial for the following reasons:

1. It complicates the tutorial, making it harder to follow.

2. I'm not sure it works as we expect it to: We use `crypto` when there is no PostHog cookie in the request. But when would there not be a PostHog cookie? Im not sure. The main scenario I can think of is when PostHog fails to initialize on the client. In this case, it wouldn't make sense to call `identify`  on the client, since PostHog isn't initialized

3. `identify`/`alias` is an advanced topic. In fact, I'm 80% sure that in the Astro and Remix tutorials we should be using `alias` instead of `identify`, and that the current implementation will cause an error – we do not allow calling `identify` for the same user with different distinct_ids (see the docs here -> https://posthog.com/docs/product-analytics/identify#what-happens-if-you-call-identify-or-alias-with-invalid-inputs )

Reply:
1. Yes, I'm fine with not doing it for this reason, we do it in enough places. 
2. There wouldn't be a PostHog cookie on first load for the user. If you want to bootstrap feature flags (or equivalent), you need to generate a user ID on the server side before the PostHog client side ever has a chance to load (and create a cookie).
3. I could be wrong, but I used `identify` instead of `alias` because the random backend distinct ID we generate is an "anonymous ID" rather than a "distinct ID". In those tutorials, we also only ever call identify once with a single client distinct ID. We (and many others) use this same pattern.

Reply:
I think once the client loads initially, we shouldn't worry about it too much. This logic can get users started, but then it is up to them to do their own identification, user ID logic. Once the client loads once, I feel like we can assume they user will get a permanent-ish ID.